src/obu.rs

```diff
@@ -2206,24 +2168,20 @@ unsafe fn parse_obus(
         r#in: &CArc<[u8]>,
         props: &Rav1dDataProps,
         gb: &mut GetBits,
-        init_bit_pos: usize,
-        init_byte_pos: usize,
-        len: usize,
     ) -> Rav1dResult {
         let hdr = parse_tile_hdr(&c.frame_hdr.as_ref().ok_or(EINVAL)?.tiling, gb);
         // Align to the next byte boundary and check for overrun.
         gb.bytealign();
-        if check_for_overrun(c, gb, init_bit_pos, len) != 0 {
+        if gb.has_error() != 0 {
             return Err(EINVAL);
         }
 
         // The current bit position is a multiple of 8
         // (because we just aligned it) and less than `8 * pkt_bytelen`
         // because otherwise the overrun check would have fired.
-        let pkt_bytelen = init_byte_pos + len;
+        let pkt_bytelen = gb.get_len();
         let bit_pos = gb.pos();
         assert!(bit_pos & 7 == 0);
-        assert!(pkt_bytelen >= bit_pos >> 3);
         let mut data = r#in.clone();
         data.slice_in_place(..pkt_bytelen);
         data.slice_in_place(bit_pos >> 3..);
```

Comment:
```suggestion
        data.slice_in_place(..gb.len());
        data.slice_in_place(gb.byte_pos()..);
```

Reply:
Do you think slicing with the index/pos first and then remaining len is easier to understand?  I thought full length and then index was simpler, but if you think it's easier to understand this way, that's fine, too.

Reply:
When slicing, I believe we most often have a pattern like `&slice[pos..][..len]`, which is what I replicated here. It might be easier to understand because it is a more common pattern. Let me know if you think otherwise.

Reply:
I had thought the other way around was simpler, but that's just me.  This way's fine, too, since you feel it's easier to understand.